### ISSUE TITLE
Possible false positives for RUF019

### ISSUE THREAD

**Issue by LordAro** [astral-sh/ruff#14011]
RUF019 can raise false positives if a class defines `__contains__` and `__getitem__` but not `get`. Alternatively, a class may implement `get` but do something entirely different with it.

(Yes, this is extremely contrived)

```
class Foo:
    def __contains__(self, a):
        return True

    def __getitem__(self, a):
        return 7


class Foo2(Foo):
    def get(self):
        return "something completely different"


f = Foo()
if "" in f and f[""]:
    pass
```

I suspect you'd be unlikely to be able to do anything about this until red_knot is done, but maybe worth documenting in the meantime? Even then might not even be able to do anything about it for the `Foo2` case.

Ruff: 0.7.1

**Comment by MichaReiser** [comment]
Thanks for reporting this issue. 

We should be able to do better with red knot because I understand from the rule description that it should only apply to dictionaries and not arbitrary other types. However, Ruff currently lacks a mechanism to detect dictionaries (it has one, but it has a too high false-negative rate).

**Comment by LordAro** [comment]
Looked into it a bit deeper, and the first example isn't nearly as contrived as first thought. Second one is a bit more contrived but maybe introducing different behaviour is worse than introducing a crash..
